swap/swap_test.go

```diff
@@ -682,3 +686,246 @@ func testDeploy(ctx context.Context, backend cswap.Backend, swap *Swap) (err err
 	}
 	return nil
 }
+
+// TestSaveAndLoadLastReceivedCheque tests if a saved last received cheque can be loaded again later using the swap functions
+func TestSaveAndLoadLastReceivedCheque(t *testing.T) {
+	swap, dir := newTestSwap(t)
+	defer os.RemoveAll(dir)
+
+	testID := newDummyPeer().Peer.ID()
+	testCheque := newTestCheque()
+
+	if err := swap.saveLastReceivedCheque(testID, testCheque); err != nil {
+		t.Fatalf("Error while saving: %s", err.Error())
+	}
+
+	returnedCheque := swap.loadLastReceivedCheque(testID)
+
+	if returnedCheque == nil {
+		t.Fatalf("Could not find saved cheque")
+	}
+
+	if returnedCheque.Amount != testCheque.Amount || returnedCheque.Beneficiary != testCheque.Beneficiary {
+		t.Fatalf("Returned cheque was different")
+	}
+}
+
+// newTestSwapAndPeer is a helper function to create a swap and a peer instance that fit together
+func newTestSwapAndPeer(t *testing.T) (*Swap, *Peer, string) {
+	swap, dir := newTestSwap(t)
+	// owner address is the beneficary (counterparty) for the peer
+	// that's because we expect cheques we receive to be signed by the address we would issue cheques to
+	peer := NewPeer(newDummyPeer().Peer, swap, nil, ownerAddress, testChequeContract)
+	// we need to adjust the owner address on swap because we will issue cheques to beneficiaryAddress
+	swap.owner.address = beneficiaryAddress
+	return swap, peer, dir
+}
+
+// TestPeerSaveAndLoadLastReceivedCheque tests if a saved last received cheque can be loaded again later using the peer functions
+func TestPeerSaveAndLoadLastReceivedCheque(t *testing.T) {
```

Comment:
Maybe this is redundant, why is it needed to also test via peer functions?

Reply:
I would keep both because the peer test wouldn't catch if the saving was implemented incorrectly (as it would still retrieve the correct cached value). The swap test on the other hand wouldn't catch an error in the caching.

Reply:
Sorry, accidentally hit the resolve button